### PR TITLE
quarto `get_chapters()` compatibility

### PR DESCRIPTION
**Pull request by cansavvy**
#### What changes are being implemented in this Pull Request?

The `get_chapters()` function is used to collect chapter information. This is used in turn by the `bookdown_to_leanpub()` function in ottrpal. 

It retrieves chapters based on the html node structure for bookdown but quarto has a different structure. 

So this is adding compatibility for the quarto structure. 

## What this needs 

The only true changes are in `R/bookdown_to_leanpub.R` the rest is styling changes. 

@howardbaek Can you double check this and see if it makes sense?

### CODE REVIEW

File: R/bookdown_to_leanpub.R
Comment (howardbaek) [inline R/bookdown_to_leanpub.R:1]: I ran `get_chapters()` with  index.html generated by the Quarto version of ottr, and got this error message:
```
Error in `dplyr::mutate()`:
ℹ In argument: `url = paste0(base_url, data_path)`.
Caused by error:
! object 'data_path' not found
```

Comment (cansavvy) [reply R/bookdown_to_leanpub.R:1]: We pair programmed on this and determined it was a simple logic issue. 

Can you test this again?

File: R/bookdown_to_leanpub.R [hunk 1]
```diff
@@ -434,13 +434,36 @@ make_embed_markdown <- function(url,
...
+  index_html <- suppressWarnings(try(xml2::read_html(bookdown_index)))
 
-  # Extract chapter nodes
+  # Extract chapter nodes the Rmd way
   nodes <- rvest::html_nodes(index_html, xpath = paste0("//", 'li[@class="chapter"]'))
 
-  if (length(nodes) > 0) {
-    # Format into a data.frame
+  # If the Rmd way didn't work, lets try the quarto way
+  if (length(nodes) < 0) {
```
Comment (howardbaek) [inline R/bookdown_to_leanpub.R:443]:
```suggestion
  if (length(nodes) < 1) {
```

The "Quarto way" `length(nodes)` is 0, so we should change the 0 to a 1.

**Comment by howardbaek** [comment]
After applying my code suggestions, I get this data frame for `chapt_data`:

```
> chapt_data
                chapt_titles data_level                       data_path
1 Introduction to the Course          1                      intro.html
2                New Chapter          2  first-section-new-chapter.html
3                New Chapter          3 second-section-new-chapter.html
4                New Chapter          4  third-section-new-chapter.html
5                 Conclusion          5                 conclusion.html
```

**Comment by cansavvy** [comment]
Thanks for these fixes @howardbaek I think this all makes sense.

**Pull request merged**